packages/tooling/__tests__/lib/parameters-to-json-schema.test.js

```diff
@@ -848,122 +764,30 @@ describe('defaults', () => {
         default: 'example default',
       };
 
-      it.each(schemaScenarios)('should pass a default of `false` [scenario: %s]', scenario => {
+      it.each(schemaScenarios)('should pass a default of `false` [scenario: %s]', async scenario => {
         const { requestBody, oas } = fixtures.generateRequestBodyDefaults('simple', scenario, { default: false });
-        expect(parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
+        expect(await parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
       });
 
       it.each(schemaScenarios)(
         'with normal non-$ref, non-inheritance, non-polymorphism cases [scenario: %s]',
-        scenario => {
+        async scenario => {
           const { requestBody, oas } = fixtures.generateRequestBodyDefaults('simple', scenario, fixtureOptions);
-          expect(parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
+          expect(await parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
         }
       );
 
-      it.each(schemaScenarios)('with simple usages of `$ref`` [scenario: %s]', scenario => {
+      it.each(schemaScenarios)('with simple usages of `$ref`` [scenario: %s]', async scenario => {
         const { requestBody, oas } = fixtures.generateRequestBodyDefaults('$ref', scenario, fixtureOptions);
-        expect(parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
+        expect(await parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
       });
 
       describe.each(polymorphismScenarios)('with usages of `%s`', refType => {
-        it.each(schemaScenarios)(`scenario: %s`, scenario => {
+        it.each(schemaScenarios)(`scenario: %s`, async scenario => {
           const { requestBody, oas } = fixtures.generateRequestBodyDefaults(refType, scenario, fixtureOptions);
-          expect(parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
+          expect(await parametersToJsonSchema({ requestBody }, oas)).toMatchSnapshot();
         });
       });
     });
-
-    describe('should comply with the `allowEmptyValue` declarative when present', () => {
```

Comment:
Since `allowEmptyValue` is actually only relevant to query params, and only when making an API request, it's not really relevant to support this kind of adhoc determination in this library, and instead this handling should happen further up the stack in `@readme/api-explorer` when we make an API request.